lib/widgets/infinite_scroll.dart

```diff
@@ -53,31 +59,39 @@ class InfiniteScroll<T> extends HookWidget {
     required this.fetcher,
     this.controller,
     this.noItems = const SizedBox.shrink(),
+    this.uniqueProp,
   }) : assert(batchSize > 0);
 
   @override
   Widget build(BuildContext context) {
     final data = useState<List<T>>([]);
+    // holds unique props of the data
+    final dataSet = useRef(HashSet<Object>());
```

Comment:
The generic parameter would then need to be moved up to the InifinteScroll forcing us to specify the generic type even when we don't care about it. In this case Object is acceptable because the only thing we want from the elements is it's hashcode to check for repeating elements (same thing is done in the `keys` parameter in many flutter_hooks)